drivers/stepper/stepper_stm32.c

```diff
@@ -0,0 +1,757 @@
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM2) {
+		if (master == TIM1) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM8) {
+			*itr = LL_TIM_TS_ITR1;
+		} else if (master == TIM3) {
+			*itr = LL_TIM_TS_ITR2;
+		} else if (master == TIM4) {
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM3) {
+		if (master == TIM1) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM2) {
+			*itr = LL_TIM_TS_ITR1;
+		} else if (master == TIM15) {
+			*itr = LL_TIM_TS_ITR2;
+		} else if (master == TIM4) {
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM4) {
+		if (master == TIM1) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM2) {
+			*itr = LL_TIM_TS_ITR1;
+		} else if (master == TIM3) {
+			*itr = LL_TIM_TS_ITR2;
+		} else if (master == TIM8) {
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM5) {
+		if (master == TIM1) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM8) {
+			*itr = LL_TIM_TS_ITR1;
+		} else if (master == TIM3) {
+			*itr = LL_TIM_TS_ITR2;
+		} else if (master == TIM4) {
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM8) {
+		if (master == TIM1) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM2) {
+			*itr = LL_TIM_TS_ITR1;
+		} else if (master == TIM4) {
+			*itr = LL_TIM_TS_ITR2;
+		} else if (master == TIM5) {
+			*itr = LL_TIM_TS_ITR3;
+		} else {
+			return -EINVAL;
+		}
+	} else if (slave == TIM12) {
+		if (master == TIM4) {
+			*itr = LL_TIM_TS_ITR0;
+		} else if (master == TIM5) {
+			*itr = LL_TIM_TS_ITR1;
+		} else {
+			return -EINVAL;
+		}
+	} else {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static void stepper_stm32_slave_timer_irq(void *arg)
+{
+	const struct stepper_stm32_config *cfg = to_config(arg);
+	struct stepper_stm32_data *data = to_data(arg);
+
+	if (LL_TIM_IsActiveFlag_UPDATE(cfg->slave_timer)) {
+		LL_TIM_ClearFlag_UPDATE(cfg->slave_timer);
+
+		if (data->pending_steps > data->max_steps) {
+			data->pending_steps -= data->max_steps;
+			LL_TIM_SetAutoReload(cfg->slave_timer,
+					     MIN(data->pending_steps,
+						 data->max_steps));
+		} else {
+			data->pending_steps = 0;
+
+			LL_TIM_DisableCounter(cfg->master_timer);
+
+			LL_TIM_DisableIT_UPDATE(cfg->slave_timer);
+			LL_TIM_DisableCounter(cfg->slave_timer);
+
+			stepper_context_complete(&data->ctx, 0);
+		}
+	}
+}
+
+static int start_move(struct device *dev, uint32_t channel, int32_t steps)
+{
+	const struct stepper_stm32_config *cfg = to_config(dev);
+	struct stepper_stm32_data *data = to_data(dev);
+
+	int32_t index;
+	const struct stepper_stm32_mconfig *mcfg;
+	struct stepper_stm32_motor_data *mdata;
+
+	index = get_motor_index(cfg, channel);
+	if (index < 0) {
+		LOG_ERR("Invalid channel: %d", channel);
+		return -EINVAL;
+	}
+
+	if (steps == 0) {
+		LOG_ERR("Number of steps can not be zero");
```

Comment:
why not? 
0 step -> do nothing.
Interest would be to have number of steps dynamically computed by application , and no need for application to check against 0.
Maybe necessary to tackle Asynchronous case.